src/interpolator_avx.rs

```diff
@@ -227,8 +216,12 @@ mod tests {
         let oversampling_factor = 256;
         let window = WindowFunction::BlackmanHarris2;
         let sincs = make_sincs::<f64>(sinc_len, oversampling_factor, f_cutoff, window);
-        let interpolator =
-            AvxInterpolator::<f64>::new(sinc_len, oversampling_factor, f_cutoff, window);
+
+        let interpolator = match AvxInterpolator::<f64>::new(sinc_len, oversampling_factor, f_cutoff, window) {
+            Ok(interpolator) => interpolator,
+            Err(..) => return,
```

Comment:
This means it can get away with an error. If new returns an error, the test will pass.

Reply:
Indeed. all though the only error it returns is if the avx interpolator isn't supported, which it isn't on the macos build agents. This was a quick and dirty way to disable (or just make the test pass) in case it's not available. I can clean it up a bit.

Reply:
you can add an assert in the Err arm, to check that avx really isn't supported. Check that with `is_x86_feature_detected!("avx")`

Reply:
So it would be a duplicate of the check inside of the constructor. What I was considering replacing it with would be something like:
```rust
let interpolator = match AvxInterpolator::<f64>::new(sinc_len, oversampling_factor, f_cutoff, window) {
    Ok(interpolator) => interpolator,
    // expected in case test is run on a platform without all the necessary CPU features.
    Err(MissingCpuFeatures(&[CpuFeature::Avx, CpuFeature::Fma])) => return,
    Err(e) => panic!("{}", e),
};
```
That way, it'll panic if there's a mismatch in the expected feature, and the check is only done in one place.

But I'll add a feature check in the test if that's what you want! libtest doesn't support disabling tests at runtime to my knowledge, so the best we can do is the same as this currently does and return.

Reply:
I was thinking something simpler, like this:
```
let interpolator = match AvxInterpolator::<f64>::new(sinc_len, oversampling_factor, f_cutoff, window) {
    Ok(interpolator) => interpolator,
    Err(_) => {
        assert!(!(is_x86_feature_detected!("avx") && is_x86_feature_detected!("fma"));
        return;
    },
};
```
Then the test doesn't depend on MissingCpuFeatures, just a simple check that it was asslowed to return an error. Right now it's a duplicate of the check in new(), but that one could change and this simpler test would still be correct.

Reply:
Done